backend/src/main/java/com/relay/service/RelayService.java

```diff
@@ -26,27 +26,27 @@ public RelayService(final RelayRepository relayRepository) {
      * 
      * @return list with all relays
      */
-    public Flux<Relay> findAll() {
+    public Iterable<Relay> findAll() {
 
         return relayRepository.findAll();
     }
 
-    public Mono<Relay> save(Relay relay) {
+    public Relay save(Relay relay) {
 
         return relayRepository.save(relay);
     }
 
-    public Mono<Relay> findOne(String id) {
+    public Optional<Relay> findOne(String id) {
 
-        return relayRepository.findById(id);
+        return relayRepository.findById(new BigInteger(id));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "BigInteger" constructor [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)